src/browser/web-view/navigation.js

```diff
@@ -155,16 +155,29 @@ export const update =
     : [ model, Effects.task(report(action.result.error)) ]
     )
   : action.type === "LocationChanged"
-  ? [ merge(model, {currentURI: action.uri})
-    , Effects.batch
-      ( [ Effects
-            .task(canGoBack(model.id))
-            .map(CanGoBackChanged)
-        , Effects
-            .task(canGoForward(model.id))
-            .map(CanGoForwardChanged)
-        ]
+  ? [ ( canGoBack != null && canGoForward != null
+      ? merge
+        ( model
+        , { currentURI: action.uri
+          , canGoBack: action.canGoBack
+          , canGoForward: action.canGoForward
+          }
+        )
+      : merge
+        ( model
+        , { currentURI: action.uri
+          }
+        )
       )
+      , Effects.batch
```

Comment:
Optional: We should not schedule `canGoBack` / `canGoForward` tasks if we already got values for them with an action.